src/OrleansRuntime/Silo/SiloHost.cs

```diff
@@ -250,6 +250,36 @@ public void WaitForOrleansSiloShutdown()
                 throw new InvalidOperationException("Cannot wait for silo " + this.Name + " due to prior initialization error");
         }
 
+		/// <summary>
+		/// Wait for this silo to shutdown or be cancelled.
+		/// </summary>
+		/// <param name="cancellationToken">Cancellation token.</param>
+		/// <remarks>
+		/// Note: This method call will block execution of current thread, 
+		/// and will not return control back to the caller until the silo is shutdown or 
+		/// an external request for cancellation has been issued.
+		/// </remarks>
+		public void WaitForOrleansSiloShutdown(CancellationToken cancellationToken)
```

Comment:
And here we can just rename the one WaitForOrleansSiloShutdown into WaitForOrleansSiloShutdown(CancellationToken cancellationToken=null) and check for null  inside and thus avoid the code dup.